packages/firebase_ml/lib/src/firebase_remote_model.dart

```diff
@@ -0,0 +1,29 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// A remote model to be downloaded to the device.
+///
+/// https://firebase.google.com/docs/reference/android/com/google/
+/// firebase/ml/common/modeldownload/FirebaseRemoteModel
+abstract class FirebaseRemoteModel {
+  /// Name associated with remote model in the Firebase console.
+  final String modelName;
+
+  /// Constructor for [FirebaseRemoteModel] that requires
+  /// a model name as an input.
+  FirebaseRemoteModel(this.modelName);
```

Comment:
If `modelName` should never be `null`, I would add an `assert` after the constructor. e.g.
```dart
FirebaseRemoteModel(this.modelName) : assert(modelName != null);
```
Same goes for all other classes.